Jenkinsfile

```diff
@@ -82,14 +83,14 @@ selectedConfigs.each { config ->
                     skipTarball: true,
                     afterBuild: { params ->
                         extraBuildSteps(params, os)
-                        maybeArchiveArtifacts(params, os)
+                        // Don't archive the Debug+ASAN artifacts
+                        if (!isDebug) {
+                            maybeArchiveArtifacts(params, os)
```

Comment:
The two levels of "maybe" is a bit odd, but the alternative would be unnecessarily passing config or isDebug into the function just so it could return again. Plus keeping it this way makes it easier to copy+paste this pattern to yet more Jenkinsfiles, most of which won't have ASAN+Debug build variants.